File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        #1. Initialize the output variables 
        outputStr = ""
        length = 0
        seen = set()
        
        #2. Two pointers 
        left = 0
        right = 0
        
        while right<len(s):
            #a. check if the character is not in right set
            if s[right] not in seen:
                seen.add(s[right])
                right+=1
                length=max(length,right-left)
            else:
                seen.remove(s[left])
                left+=1
        return length

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        //1. Output variables
        Set<Character> seen = new HashSet<>();
        int outputResult = 0;
        
        //2. Pointers
        int left = 0;
        int right = 0;
        
        //3. While loop as long as right<length
        while(right<s.length()){
            //a. is it in the seen 
            if (seen.contains(s.charAt(right))){
                seen.remove(s.charAt(left));
                left+=1;
                
            }else{
                seen.add(s.charAt(right));
                right+=1;
                outputResult = Math.max(outputResult, right-left);
            }
        }
        return outputResult;
        
    }
}